src/ipc/ipc4/helper.c

```diff
@@ -342,6 +343,17 @@ int ipc_pipeline_free(struct ipc *ipc, uint32_t comp_id)
 	if (!cpu_is_me(ipc_pipe->core))
 		return ipc4_process_on_core(ipc_pipe->core, false);
 
+	if (ipc_pipe->pipeline->init_time) {
+		/*
+		 * This pipeline must not be destroyed before a minimum time
+		 * since its creation has passed
+		 */
+		life_time = k_cyc_to_ms_near64(sof_cycle_get_64() - ipc_pipe->pipeline->init_time);
+		pipe_dbg(ipc_pipe->pipeline, "Extend pipeline life beyond %llu", life_time);
+		if (life_time < CONFIG_LIBRARY_PIPELINE_FORCE_MIN_LIFETIME)
+			k_msleep(CONFIG_LIBRARY_PIPELINE_FORCE_MIN_LIFETIME - life_time);
```

Comment:
This still looks quite odd. I guess there is some race in start and stopping the pipeline tasks, but this seems awfully brittle way to address it.

Reply:
#9371